backend/src/routes/ontologies.ts

```diff
@@ -84,26 +94,26 @@ const getSubGoalsfromSDG = async (req, res) => {
   }
 };
 
-const regexSearch = async (req, res) => {
+const regexSearch = async (req: RegexRequest, res: NodeArrayResponse) => {
   try {
     const searchTerm = req.query.search;
     const limitResults = req.query.limit;
-    const data = await getClassesByString(searchTerm, limitResults);
-
+    verifyRequestQueryParams(searchTerm);
+    const data = await getClassesByString(searchTerm!, limitResults);
```

Comment:
Can `searchTerm` be guaranteed here or should it be null-checked first?

Reply:
verifyRequestQueryParams checks for null values and throws an exception if null or undefined.